src/libvncserver/sockets.c

```diff
@@ -715,9 +716,10 @@ rfbReadExactTimeout(rfbClientPtr cl, char* buf, int len, int timeout)
 #endif
             FD_ZERO(&fds);
             FD_SET(sock, &fds);
+	    exceptfds = fds;
             tv.tv_sec = timeout / 1000;
             tv.tv_usec = (timeout % 1000) * 1000;
-            n = select(sock+1, &fds, NULL, &fds, &tv);
+            n = select(sock+1, &fds, NULL, &exceptfds, &tv);
```

Comment:
…and if `select()` sets a bit in `exceptfds` where is it checked in the following code?

Reply:
Good point. I might have missed something there, but afaict nobody is checking `fds` currently.

Reply:
Okay. I am not familiar with the semantics of `select()` here, but wouldn't it be the same as passing `NULL` if the caller is not interested in the file descriptors anyway?

Reply:
According to the manpage, `NULL` is equivalent to an empty set. So this is different, as it specifies a set with one fd set.

Regarding the checking of the result of `select()`: I can not claim to fully understand the code, but only a single fd is set in fds (`FD_SET(sock, &fds);`). I guess this fd is probed afterwards, so that the result of `select()` is not needed.